### PR TITLE
Fixed #33783 -- Added IsEmpty GIS database function and __isempty lookup on PostGIS.

### PR DESCRIPTION
**Pull request by claudep**

### CODE REVIEW

File: tests/gis_tests/geoapp/test_functions.py [hunk 1]
```diff
@@ -371,6 +371,17 @@ def test_intersection(self):
             else:
                 self.assertIs(c.inter.empty, True)
 
+    @skipUnlessDBFeature("supports_empty_geometries")
```
Comment (felixxm) [inline tests/gis_tests/geoapp/test_functions.py:374]: I can imagine a backend which supports empty geometries but doesn't have `IsEmpty` function, so I added `IsEmpty` to the `unsupported_functions`.

File: django/contrib/gis/db/models/functions.py [hunk 2]
```diff
@@ -381,6 +381,12 @@ class Intersection(OracleToleranceMixin, GeomOutputGeoFunc):
     geom_param_pos = (0, 1)
 
 
+class IsEmpty(GeoFunc):
+    lookup_name = "isempty"
+    output_field = BooleanField()
+    arity = 1
```
Comment (felixxm) [inline django/contrib/gis/db/models/functions.py:387]: We could support `__isempty` lookup.

**Review by felixxm — COMMENTED** [review]
@claudep Thanks :+1:

I added support for `__isempty` lookup and made some doc changes.

**Comment by claudep** [comment]
Thanks, looks good! However, the link between this patch and the test_make_line failures is not obvious :grimacing: Tell me if you need help.

**Comment by felixxm** [comment]
> Thanks, looks good! However, the link between this patch and the test_make_line failures is not obvious grimacing Tell me if you need help.

It seems that the new tests have broken the shaky balance of the default ordering :) (see ticket-23842.)

**Comment by felixxm** [comment]
> > Thanks, looks good! However, the link between this patch and the test_make_line failures is not obvious grimacing Tell me if you need help.
> 
> It seems that the new tests have broken the shaky balance of the default ordering :) (see [ticket-23842](https://code.djangoproject.com/ticket/23842).)

I've tried to fix it, see #16419.

**Pull request merged**